diff --git a/gpu_checker.py b/gpu_checker.py
old mode 100644
new mode 100755
index ed9a7f8..75a80d1
--- a/gpu_checker.py
+++ b/gpu_checker.py
@@ -1,3 +1,4 @@
+#!/usr/bin/env python3
 """
 Simon Leary
 6/30/2022
@@ -322,7 +323,7 @@ def init_config():
         with open('gpu_checker_config.ini', 'w', encoding='utf-8') as config_file:
             config_file.write(CONFIG_PREPEND)
             config.write(config_file)
-        os.chmod('gpu_checker_config.ini', 0o700) # 0o means octal digits
+        os.chmod('gpu_checker_config.ini', 0o600) # 0o means octal digits
     return config
 
 if __name__=="__main__":
@@ -347,11 +348,16 @@ if __name__=="__main__":
                                  include_nodes, exclude_nodes):
                 continue # next node
             # else:
-            gpu_works, check_report = check_gpu(node)
-            if gpu_works:
-                LOG.info(f"{node} works")
-                time.sleep(post_check_wait_time_s)
-                continue # next node
+            try:
+                gpu_works, check_report = check_gpu(node)
+                if gpu_works:
+                    LOG.info(f"{node} works")
+                    time.sleep(post_check_wait_time_s)
+                    continue # next node
+            except Exception as e:
+                # SSH failed
+                LOG.error(f"{node} check failed: {e}")
+                continue  # next node
             # else:
             LOG.error(f"{node} doesn't work!")
             if do_drain_nodes:
